chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomService.kt

```diff
@@ -180,7 +182,17 @@ internal class RoomService(
                 messageLimit = messageLimit,
                 roomId = roomId,
                 cursorService = cursorsService,
-                membershipConsumer = { roomStore.applyMembershipEvent(roomId, it).map(::enrichEvent).forEach(emit) },
+                membershipConsumer = {
+                    if (it is MembershipSubscriptionEvent.InitialState
+                            && !roomsPassedInitialState.contains(roomId)) {
+                        //if it's the first initial state event we don't want to emit the callbacks for
+                        //people who have entered the room
+                        roomStore.applyMembershipEvent(roomId, it).map(::enrichEvent)
```

Comment:
This is a very minor sytlistic point, comment rather than request: generally one `map`s if the results are useful (and the mapping is a pure function, which is not true here and is what confused us!) and uses `forEach` if the side-effects are the reason for the iteration and the return value would be discarded.

Reply:
Yeah! i think I agree. I guess there's no point noting this down, but keep it in mind when we re-write the SDK?